tests/test_io.py

```diff
@@ -2704,21 +2705,27 @@ def test_magicc_data_append(mock_read_and_return_metadata_df):
         tdf_append_df,
         tdf_append_columns,
     )
-    mdata.append(tfilepath)
+    if inplace:
+        mdata.append(tfilepath, inplace=inplace)
+        res = mdata
+    else:
+        res = mdata.append(tfilepath, inplace=inplace)
 
     mock_read_and_return_metadata_df.assert_called_with(tfilepath)
 
+    assert isinstance(res, MAGICCData)
+
     expected_metadata = deepcopy(tmetadata_init)
     expected_metadata.update(tmetadata_append)
-    assert mdata.metadata == expected_metadata
+    assert res.metadata == expected_metadata
 
     expected = pd.concat([tdf_init, tdf_append])
     expected.columns = pd.Index(
         [datetime.datetime(tindex_yr, 1, 1, 0, 0, 0)], dtype="object"
     )
 
     pd.testing.assert_frame_equal(
-        mdata.timeseries(),
+        res.timeseries(),
         expected.sort_index().reorder_levels(mdata.timeseries().index.names),
         check_like=True,
     )
```

Comment:
Add a check to check that the original mdata hasn't changed for inplace=False

Reply:
how do we do comparisons for `MAGICCData` objects?

Reply:
Perhaps keep a copy of the original timeseries, do the append and check that they are the same
```
orig_timeseries = mdata.timeseries().copy()

...

if not inplace:
      pd.testing.assert_frame_equal(
        orig_timeseries,
        mdata.timeseries(),
        check_like=True,
    )
```

Perhaps ScmDataFrame needs an __eq__ operator or a helper?

Reply:
> Perhaps ScmDataFrame needs an **eq** operator or a helper?

Having just read about these I am nervous, equality is a tricky thing to define. I like leaving it up to the users to decide what is equal enough (just `timeseries`, `timeseries` and `metadata` etc.)